File: 11. Container With Most Water/main.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let n = height.length;
  let i = 0,
    j = n - 1;
  let maxVolume = 0;

  while (i < j) {
    let volume = (j - i) * (height[i] < height[j] ? height[i] : height[j]);
    if (volume > maxVolume) maxVolume = volume;
    height[i] < height[j] ? ++i : --j;
  }
  return maxVolume;
};

File: 11. Container With Most Water/main.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        output, lp, rp,  = 0, 0, len(height) - 1

        while lp < rp:
            left, right = height[lp], height[rp]
            output = max(output, (rp-lp) * min(left, right))

            if left <= right:
                lp += 1
            else:
                rp -= 1

        return output